runtime/common/Logger.h

```diff
@@ -53,7 +53,14 @@ void debug(const std::string_view msg);
   template <typename... Args>                                                  \
   NAME(const std::string_view, Args &&...) -> NAME<Args...>;
 
+#if (defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER))
```

Comment:
Can we remove this around the macro invocation and just add `-Wrestrict` to the block in `FmtCore.h`?

Reply:
We can. I haven't looked closely why this warning is popping up here but nowhere else.

Reply:
done